config/metric.go

```diff
@@ -258,3 +269,136 @@ func (rws readerWithServer) Shutdown(ctx context.Context) error {
 		rws.server.Shutdown(ctx),
 	)
 }
+
+func view(v View) (sdkmetric.View, error) {
+	if v.Selector == nil {
+		return nil, errors.New("view: no selector provided")
+	}
+
+	inst, err := instrument(*v.Selector)
+	if err != nil {
+		return nil, err
+	}
+
+	return sdkmetric.NewView(inst, stream(v.Stream)), nil
+}
+
+func instrument(vs ViewSelector) (sdkmetric.Instrument, error) {
+	kind, err := instrumentKind(vs.InstrumentType)
+	if err != nil {
+		return sdkmetric.Instrument{}, fmt.Errorf("view_selactor: %w", err)
+	}
+	return sdkmetric.Instrument{
+		Name: strOrEmpty(vs.InstrumentName),
+		Unit: strOrEmpty(vs.Unit),
+		Kind: kind,
+		Scope: instrumentation.Scope{
+			Name:      strOrEmpty(vs.MeterName),
+			Version:   strOrEmpty(vs.MeterVersion),
+			SchemaURL: strOrEmpty(vs.MeterSchemaUrl),
+		},
+	}, nil
+}
+
+func stream(vs *ViewStream) sdkmetric.Stream {
+	if vs == nil {
+		return sdkmetric.Stream{}
+	}
+
+	return sdkmetric.Stream{
+		Name:            strOrEmpty(vs.Name),
+		Description:     strOrEmpty(vs.Description),
+		AttributeFilter: attributeFilter(vs.AttributeKeys),
+		Aggregation:     aggregation(vs.Aggregation),
+	}
+}
+
+func attributeFilter(attributeKeys []string) attribute.Filter {
+	var attrKeys []attribute.Key
+	for _, attrStr := range attributeKeys {
+		attrKeys = append(attrKeys, attribute.Key(attrStr))
+	}
+	return attribute.NewAllowKeysFilter(attrKeys...)
+}
+
+func aggregation(aggr *ViewStreamAggregation) sdkmetric.Aggregation {
+	if aggr == nil {
+		// TODO: Understand what to set here.
+		return nil
+	}
+
+	if aggr.Base2ExponentialBucketHistogram != nil {
+		return sdkmetric.AggregationBase2ExponentialHistogram{
+			MaxSize:  int32(intOrZero(aggr.Base2ExponentialBucketHistogram.MaxSize)),
+			MaxScale: int32(intOrZero(aggr.Base2ExponentialBucketHistogram.MaxScale)),
+			// Need to negate because config has the positive action RecordMinMax.
+			NoMinMax: !boolOrFalse(aggr.Base2ExponentialBucketHistogram.RecordMinMax),
+		}
+	}
+	if aggr.Default != nil {
+		// TODO: Understand what to set here.
```

Comment:
yeah.... but it is not working... The reason is that I need to know the `InstrumentKind` which I don't always know, since the Instrument may or may not have the kind.

Reply:
The problem is that the view.instrument may match multiple instrument kinds so cannot pick one specific aggregation.